interfaces/Portalicious/src/app/domains/registration/registration.api.service.ts

```diff
@@ -85,6 +88,61 @@ export class RegistrationApiService extends DomainApiService {
     });
   }
 
+  changeStatus({
+    projectId,
+    paginateQuery,
+    status,
+    messageData,
+    dryRun = true,
+  }: {
+    projectId: Signal<number>;
+    paginateQuery: PaginateQuery | undefined;
+    status: RegistrationStatusEnum;
+    messageData?: SendMessageData | undefined;
+    dryRun: boolean;
+  }) {
+    let body: RegistrationStatusPatchDto;
+    body = {
+      status,
+    };
+    if (messageData && 'customMessage' in messageData) {
+      body = {
+        ...body,
+        message: messageData.customMessage,
+        status,
+      };
+    } else if (messageData) {
+      body = {
+        ...body,
+        messageTemplateKey: messageData.messageTemplateKey,
+      };
+    }
```

Comment:
```suggestion
    if (messageData && 'customMessage' in messageData) {
      body.message = messageData.customMessage;
    } else if (messageData) {
      body.messageTemplateKey = messageData.messageTemplateKey;
    }
```

Maybe slightly easier to read?